lib/facter/udevadm.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# Fact: udevadm
+#
+# Purpose:
+#   Provide information about the udevadm utilty.
+#
+# Resolution:
+#   Check if the kernel is linux.
+#
+# Caveats:
+#
+Facter.add(:udevadm) do
+  confine kernel: :linux
+
+  path = Facter::Util::Resolution.which('udevadm')
+
+  if path
```

Comment:
I think in general it's bad to execute code outside of `setcode`. It doesn't respect the `confine` bits and gets executed on every OS.

Reply:
:facepalm:

Reply:
This should be addressed and I've converted to using `Facter::Core::Execution` as well.